spec-tests/src/test/java/org/candlepin/spec/InactiveConsumerCleanerJobTest.java

```diff
@@ -0,0 +1,124 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class InactiveConsumerCleanerJobTest extends SpecTestFixture {
+
+    private static final String JOB_KEY = "InactiveConsumerCleanerJob";
+    private static final int DEFAULT_LAST_CHECKED_IN_RETENTION_IN_DAYS = 397;
+
+    private ConsumerClient consumerClient;
+    private OwnerClient ownerClient;
+    private JobsClient jobsClient;
+    private ConsumerTypeClient consumerTypeClient;
+
+    private OwnerDTO owner;
+    private ConsumerTypeDTO consumerType;
+
+    @BeforeEach
+    private void beforeEach() throws ApiException {
+        consumerClient = getClient(ConsumerClient.class);
+        ownerClient = getClient(OwnerClient.class);
+        jobsClient = getClient(JobsClient.class);
+        consumerTypeClient = getClient(ConsumerTypeClient.class);
+
+        if (owner == null) {
+            owner = ownerClient.createOwner(new OwnerDTOBuilder().build());
+        }
+
+        if (consumerType == null) {
+            ConsumerTypeDTO newType = new ConsumerTypeDTO();
+            newType.setManifest(false);
+            newType.setLabel(Util.randomString("Test Type"));
+            consumerType = consumerTypeClient.createConsumerType(newType);
+        }
+    }
+
+    @Test
+    public void shouldDeleteInactiveConsumers() throws Exception {
+        Instant inactiveTime = Instant.now().minus(DEFAULT_LAST_CHECKED_IN_RETENTION_IN_DAYS + 10,
+            ChronoUnit.DAYS);
+        Instant activeTime = Instant.now().minus(DEFAULT_LAST_CHECKED_IN_RETENTION_IN_DAYS - 10,
+            ChronoUnit.DAYS);
+
+        ConsumerDTO inactiveConsumer = buildConsumer(inactiveTime.atOffset(ZoneOffset.UTC));
+        ConsumerDTO activeConsumer = buildConsumer(activeTime.atOffset(ZoneOffset.UTC));
+        ConsumerDTO activeConsumer2 = buildConsumer(activeTime.atOffset(ZoneOffset.UTC));
+
+        inactiveConsumer = consumerClient.createConsumer(inactiveConsumer, null, owner.getKey(), null, false);
+        activeConsumer = consumerClient.createConsumer(activeConsumer, null, owner.getKey(), null, false);
+        activeConsumer2 = consumerClient.createConsumer(activeConsumer2, null, owner.getKey(), null, false);
+
+        AsyncJobStatusDTO status = jobsClient.scheduleJob(JOB_KEY, null);
+        boolean isComplete = jobsClient.waitForJobToComplete(status.getId());
+        assertTrue(isComplete);
+
+        // Verify that the inactive consumer has been deleted.
+        final String inactiveConsumer1Uuid = inactiveConsumer.getUuid();
+        ApiException thrown1 = Assertions.assertThrows(ApiException.class, () -> {
+            consumerClient.getConsumer(inactiveConsumer1Uuid);
+        });
+        assertEquals(410, thrown1.getCode());
```

Comment:
Take a look at StatusCodeAssertions a collection of status code related assetions.

Reply:
Updated!